device-connectors/src/testflinger_device_connectors/devices/maas2/maas2.py

```diff
@@ -239,6 +239,47 @@ def _run_tpm_clear_cmd(self):
             return True
         return False
 
+    def run_maas_cmd_with_retry(self, cmd, max_retries=5, backoff_start=60):
+        """
+        Run maas command with retries on failure
+
+        :param cmd:
+            MAAS command to be run
+        :param max_retries:
+            Maximum amount of times to retry the command on failure
+        :param backoff_start:
+            Initial time in seconds to sleep after failure
+        """
+        retry_count = 0
+        while True:
+            proc = subprocess.run(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                check=False,
+            )
+            if proc.returncode:
+                if retry_count > max_retries:
+                    self._logger_error(
+                        (
+                            f"maas error running: {' '.join(cmd)}"
+                            "maximum retries reached"
+                        )
+                    )
+                    raise ProvisioningError(proc.stdout.decode())
+
+                timeout = backoff_start * 2**retry_count
+                self._logger_warning(
+                    (
+                        f"maas error running: {' '.join(cmd)}"
+                        f"trying again in {timeout} seconds"
+                    )
+                )
+                time.sleep(timeout)
+                retry_count += 1
+            else:
+                return proc
```

Comment:
Now that this is a bit simpler, I think we can reduce the number of indents needed here:

```suggestion
            if proc.returncode == 0:
                return proc
            
            if retry_count > max_retries:
                self._logger_error(
                    (
                        f"maas error running: {' '.join(cmd)}"
                        "maximum retries reached"
                    )
                )
                raise ProvisioningError(proc.stdout.decode())

            timeout = backoff_start * 2**retry_count
            self._logger_warning(
                (
                    f"maas error running: {' '.join(cmd)}"
                    f"trying again in {timeout} seconds"
                )
            )
            time.sleep(timeout)
            retry_count += 1
```